### ISSUE TITLE
Emails are triggered to the approver for each expense added to a report with instant submit

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#50825]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.49-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C05LX9D6E07/p1728990449040979

## Action Performed:
1. Have a policy with instant submit
2. As a submitter submit an expense in a report to approver
3. Add another expense to the report

## Expected Result:
Emails shouldn't be received to the approver for every new expense added to the report. Should trigger emails when the report is submitted and not for subsequent new expenses on it
## Actual Result:
Emails are received for approver for every individual expense created on a report that was then submitted to the approver
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
![CleanShot 2024-10-15 at 13 06 13@2x](https://github.com/user-attachments/assets/81883d0d-c727-4be9-9977-5930f0bd36ca)

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by trjExpensify** [comment]
Hold on progressing this for a sec, discussion in thread: https://expensify.slack.com/archives/C05LX9D6E07/p1729069163722329?thread_ts=1728990449.040979&cid=C05LX9D6E07

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by danieldoglas** [comment]
still on hold.

**Comment by JmillsExpensify** [comment]
Still on hold

**Comment by danieldoglas** [comment]
Still on hold. I'll remove myself from this issue, and we can assign engineering again when we remove hold

**Comment by JmillsExpensify** [comment]
We'll hopefully be able to remove the hold soon, as we're just working through bugs/polish for the related issue.

**Comment by trjExpensify** [comment]
This should be fixed as we don't notify for every request preview added to the report anymore.

**Comment by mallenexpensify** [comment]
Removed `AutoAssignerNewDotQuality`, removed hold in title, added `retest-weekly` and[ posted in #qa ](https://expensify.slack.com/archives/C9YU7BX5M/p1739742713371459)asking for a retest
> Can I get a retest here, if you're unable to reproduce, plz close. Thx

**Comment by JmillsExpensify** [comment]
Sounds good, thanks!

**Comment by kavimuru** [comment]
Bug is still reproducible.

https://github.com/user-attachments/assets/80720ed1-f6a7-425b-9f35-f41a416a6634

**Comment by mallenexpensify** [comment]
@trjExpensify I don't think QA team actively monitors GHs for tags, I [posted in the #qa thread ](https://expensify.slack.com/archives/C9YU7BX5M/p1739930077247179?thread_ts=1739742713.371459&cid=C9YU7BX5M)to ask Kavi for 👀.

**Comment by trjExpensify** [comment]
Thanks! 

**TL;DR** it's the `"false"` string that's the issue. I think we need to up the priority of https://github.com/Expensify/Expensify/issues/447548 as this is causing OldDot emails to trigger to NewDot users, and NewDot redirects to be ignored. CC: @Julesssss 

____
Approver: applausetester+bp2401pr@applause.expensifail.com. AccountID: 19037120

```
        "tryNewDot": {
            "classicRedirect": {
                "completedHybridAppOnboarding": "true",
                "dismissed": "false",
                "dismissedReasons": [
                    {
                        "reason": "preferClassic - fghxchf",
                        "timestamp": "2025-02-02 11:17:53.155"
                    }
                ],
                "timestamp": "2025-02-13 13:59:52"
            },
```

That account has `"dismissed": "false",`set as a a string, not boolean. @Julesssss has an issue to look into it [here](https://github.com/Expensify/Expensify/issues/447548).

PreviousIy, that has answered why the approver is getting the OldDot emails. As for if that's the answer for why an email is being triggered for every expense added to the report...?

Looking at reportID: 4460475265417357. ([Logs here](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%224460475265417357%22%20AND%20timestamp%3A%5B2025-02-17T00%3A00%20TO%202025-02-17T23%3A59%5D))

The comments on the report came after the expenses were inserted, so I don't think it's that, unless some comments were deleted or something which it doesn't look like it in the logs.
<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/25e34b50-2a4d-44a2-83dd-d9814257f2b4" />

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/b0582cf8-aae3-499c-9fec-697786cbbcdd" />

So let's try reproducing the conditions of the account... 


**Before making my dismissed value a string.** 
Just the one email, lovely. ✅ 

<img width="1211" alt="Image" src="https://github.com/user-attachments/assets/a3b08d58-5139-4682-b1c5-19b765a733d4" />

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/fb3a1b58-8bb2-4887-891c-0ed08c29720b" />

**After making my dismissed value a string.** 
OldDot emails sent, and an email for every expense added to the report. Boo! ❌ 

<img width="1222" alt="Image" src="https://github.com/user-attachments/assets/64a67017-e410-4786-9b0b-0ca7666f5c37" />

<img width="1511" alt="Image" src="https://github.com/user-attachments/assets/98658073-680c-41d2-9627-dd483a3c2a0f" />

**Repro Steps**
1. Go to new.expensify.com 
2. Sign-up for a new account 
3. Create a workspace > go to workflows > disable delay submit > go to members > invite a member
4. Go to Settings > Domains 
5. Open the JS console on OldDot and paste in the below snippet to get your false value to a string: 

```
const nvp = NVP.get('tryNewDot') ?? {classicRedirect: {}};
nvp.classicRedirect.dismissed = "false";
NVP.set('tryNewDot', nvp);
```

6. As the invited member, submit **two** expenses to the approver on NewDot. 
7. As the approver, open your email.
8. Observe the multiple emails for each of the expenses submitted in step 6.

**Comment by Julesssss** [comment]
Oh damn, nice find Tom.

The steps aren't time consuming, but have just been hard to prioritise. This is a good reason to get it done.

**Comment by trjExpensify** [comment]
Yeah, agreed. 👍

**Comment by mallenexpensify** [comment]
> I think we need to up the priority of https://github.com/Expensify/Expensify/issues/447548

I'm seeing that listed as `Done` for priority, would it be helpful to update that? 

<img width="361" alt="Image" src="https://github.com/user-attachments/assets/545ccb05-0d61-4ad5-8de6-294bc4daa86a" />

For this issue, should we put it on hold pending https://github.com/Expensify/Expensify/issues/447548 ?

For status here, it should either be one of the below.  
HIGH -  UX Reliability, API Reliability and Performance issues that are part of main flows and affect a subset of users. 
LOW - Not very noticeable UX Reliability, API Reliability or Performance issues that only affect a single user or subset of users

Can someone more familiar with this update the status on comment on what you think it should be?  If you're unsure, make it HIGH so we can keep 👀 on .  Thx

**Comment by trjExpensify** [comment]
Yeah, we fixed the bug in that issue, and now we're just putting together a CQ to clean-up historical setting of `"false"` || `"true"` https://github.com/Expensify/Expensify/issues/447548#issuecomment-2718888077

So we can close this.
